src/discussions/discussions-home/DiscussionsHome.jsx

```diff
@@ -94,65 +98,82 @@ const DiscussionsHome = () => {
               {!enableInContextSidebar && (
                 <NavigationBar />
               )}
-              <PostActionsBar />
+              {(isEnrolled || enableInContextSidebar) && <PostActionsBar />}
             </div>
             <DiscussionsRestrictionBanner />
           </div>
+
           {provider === DiscussionProvider.LEGACY && (
-            <Suspense fallback={(<Spinner />)}>
-              <Routes>
-                {[
-                  ROUTES.TOPICS.CATEGORY,
-                  ROUTES.TOPICS.CATEGORY_POST,
-                  ROUTES.TOPICS.CATEGORY_POST_EDIT,
-                  ROUTES.TOPICS.TOPIC,
-                  ROUTES.TOPICS.TOPIC_POST,
-                  ROUTES.TOPICS.TOPIC_POST_EDIT,
-                ].map((route) => (
-                  <Route
-                    key={route}
-                    path={route}
-                    element={<LegacyBreadcrumbMenu />}
-                  />
-                ))}
-              </Routes>
-            </Suspense>
+          <Suspense fallback={(<Spinner />)}>
+            <Routes>
+              {[
+                ROUTES.TOPICS.CATEGORY,
+                ROUTES.TOPICS.CATEGORY_POST,
+                ROUTES.TOPICS.CATEGORY_POST_EDIT,
+                ROUTES.TOPICS.TOPIC,
+                ROUTES.TOPICS.TOPIC_POST,
+                ROUTES.TOPICS.TOPIC_POST_EDIT,
+              ].map((route) => (
+                <Route
+                  key={route}
+                  path={route}
+                  element={<LegacyBreadcrumbMenu />}
+                />
+              ))}
+            </Routes>
+          </Suspense>
+          )}
+          {!isEnrolled && courseStatus === LOADED && (
+          <Suspense fallback={(<Spinner />)}>
+            <Routes>
+              {ALL_ROUTES.map((route) => (
+                <Route
+                  key={route}
+                  path={route}
+                  element={(<ContentUnavailable subTitleMessage={messages.contentUnavailableSubTitle} />)}
+                />
+              ))}
+            </Routes>
+          </Suspense>
           )}
+          {((isEnrolled && courseStatus === LOADED) || enableInContextSidebar) && (
```

Comment:
We do not need this check. These routes will only work when the user is enrolled.

Reply:
we need to add checks to avoid flicker, i added tertiary operator for this case.